cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java

```diff
@@ -380,13 +381,108 @@ private void convertACLineSegmentsToLines(Context context, Set<String> delayedBo
             if (c.valid()) {
                 String node = c.boundaryNode();
                 if (node != null && !context.config().convertBoundary()) {
-                    context.boundary().addAcLineSegmentAtNode(line, node);
-                    delayedBoundaryNodes.add(node);
+                    String lineContainerId = line.getId("Line");
+                    if (lineContainerId != null) {
+                        delayedLines.computeIfAbsent(lineContainerId, l -> new PropertyBags()).add(line);
+                    } else {
+                        context.boundary().addAcLineSegmentAtNode(line, node);
+                        delayedBoundaryNodes.add(node);
+                    }
                 } else {
                     c.convert();
                 }
             }
         }
+        Map<String, PropertyBags> linesCopy = new HashMap<>(delayedLines);
+        linesCopy.forEach((lineContainerId, p) -> {
+            boolean onlyOneLineAtBoundary = p.size() == 1;
+            boolean twoLinesAtBoundary = p.size() == 1;
+            if (onlyOneLineAtBoundary || twoLinesAtBoundary) {
+                p.forEach(line -> {
+                    ACLineSegmentConversion c = new ACLineSegmentConversion(line, context);
+                    context.boundary().addAcLineSegmentAtNode(line, c.boundaryNode());
+                    delayedBoundaryNodes.add(c.boundaryNode());
+                });
+                delayedLines.remove(lineContainerId, p);
+            }
+        });
+    }
+
+    private void convertLine(Context context, String lineId, PropertyBags lineSegments) {
```

Comment:
I am not sure that we need the approach of delaying the conversion of `ACLineSegments` found in `cim:Line` containers. For me, they should be converted using the current existing considerations (could be located at boundaries, could be merged with others at boundary, ...). The only consideration that we must add is that one of the ends will be a fictitious voltage level, that we create as we need it. We should create a fictitious voltage level for every `ConnectivityNode` inside the cim:Line; all the `ACLineSegments` sharing the same connectivity node in the CGMES model will be mapped to IIDM Lines in the same fictitious voltage level.